crates/aptos/src/move_tool/stored_package.rs

```diff
@@ -134,6 +134,62 @@ impl<'a> CachedPackageMetadata<'a> {
         }
         Ok(())
     }
+
+    pub fn verify(&self, package_metadata: &PackageMetadata) -> anyhow::Result<()> {
+        let self_metadata = self.metadata;
+
+        if self_metadata.name != package_metadata.name {
+            bail!(
+                "Package name doesn't match {} : {}",
+                package_metadata.name,
+                self_metadata.name
+            )
+        } else if self_metadata.deps != package_metadata.deps {
+            bail!(
+                "Dependencies don't match {:?} : {:?}",
+                package_metadata.deps,
+                self_metadata.deps
+            )
+        } else if self_metadata.modules != package_metadata.modules {
+            bail!(
+                "Modules don't match {:?} : {:?}",
+                package_metadata.modules,
+                self_metadata.modules
+            )
+        } else if self_metadata.manifest != package_metadata.manifest {
+            bail!(
+                "Manifest doesn't match {:?} : {:?}",
+                package_metadata.manifest,
+                self_metadata.manifest
+            )
+        } else if self_metadata.upgrade_policy != package_metadata.upgrade_policy {
+            bail!(
+                "Upgrade policy doesn't match {:?} : {:?}",
+                package_metadata.upgrade_policy,
+                self_metadata.upgrade_policy
+            )
+        } else if self_metadata.upgrade_number != package_metadata.upgrade_number {
+            bail!(
+                "Upgrade number doesn't match {:?} : {:?}",
+                package_metadata.upgrade_number,
```

Comment:
The upgrade number should not be compared. It is actually 0 for a locally build package and will be assigned at the moment the package is published on chain.